components/eam/src/physics/cam/ndrop.F90

```diff
@@ -306,190 +306,166 @@ subroutine dropmixnuc( &
+   real(r8), allocatable :: raercol_cw(:,:,:) ! same as raercol but for cloud-borne phase [# or kg/kg]
+   real(r8), allocatable :: naermod(:)  ! aerosol number concentration [#/m^3]
+   real(r8), allocatable :: hygro(:)    ! hygroscopicity of aerosol mode [dimensionless]
+   real(r8), allocatable :: vaerosol(:) ! aerosol volume conc [m^3/m^3]
+   real(r8), allocatable :: fn(:)              ! activation fraction for aerosol number [fraction]
+   real(r8), allocatable :: fm(:)              ! activation fraction for aerosol mass [fraction]
+   real(r8), allocatable :: fluxn(:)           ! number  activation fraction flux [m/s]
+   real(r8), allocatable :: fluxm(:)           ! mass    activation fraction flux [m/s]
 
-   real(r8), allocatable :: fn(:)              ! activation fraction for aerosol number
-   real(r8), allocatable :: fm(:)              ! activation fraction for aerosol mass
-
-   real(r8), allocatable :: fluxn(:)           ! number  activation fraction flux (cm/s)
-   real(r8), allocatable :: fluxm(:)           ! mass    activation fraction flux (cm/s)
-   real(r8)              :: flux_fullact(pver) ! 100%    activation fraction flux (cm/s)
    !     note:  activation fraction fluxes are defined as
-   !     fluxn = [flux of activated aero. number into cloud (#/cm2/s)]
-   !           / [aero. number conc. in updraft, just below cloudbase (#/cm3)]
-
+   !     fluxn = [flux of activated aero. number into cloud [#/m^2/s]]
+   !           / [aero. number conc. in updraft, just below cloudbase [#/m^3]]
 
    real(r8), allocatable :: coltend(:,:)       ! column tendency for diagnostic output
    real(r8), allocatable :: coltend_cw(:,:)    ! column tendency
-   real(r8) :: ccn(pcols,pver,psat)    ! number conc of aerosols activated at supersat
-   integer :: ccn3d_idx
-   real(r8), pointer :: ccn3d(:, :)
+   type(ptr2d_t), allocatable :: qqcw(:)     ! cloud-borne aerosol mass, number mixing ratios [# or kg/kg]
 
+   integer, save :: count_submix(100)
+   integer  :: lchnk               ! chunk identifier
+   integer  :: ncol                ! number of columns
+   integer  :: mm                  ! local array index for MAM number, species
+   integer  :: km1, kp1            ! level index -1, +1
+   integer  :: nnew, nsav, ntemp   ! indices for old, new time levels in substepping
+   integer  :: lptr
+   integer  :: nsubmix, nsubmix_bnd  ! number of substeps and bound
+   integer  :: phase                 ! phase of aerosol
+   integer  :: ccn3d_idx   ! index of ccn3d in pbuf
+   integer  :: icol        ! column index
+   integer  :: imode       ! mode index
+   integer  :: kk          ! level index
+   integer  :: lspec      ! species index for given mode
+   integer  :: lsat       !  level of supersaturation
+   integer  :: isub       ! substep index
+   integer  :: spc_idx, num_idx  ! species, number indices  
+
+! NOTE FOR C++ PORT:  Below fields are not used in our tests since do_aerocom_ind3 = .false. by default.
```

Comment:
As discussed in Wednesday, I think we can remove these code in our refactoring side. Anyway, they are not tested and will not be ported into C++. We always have the original code for backup.

Reply:
Thanks -- all the code invoked by do_aerocom_ind3 = .true. has been removed.